packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketPaymentMethod.tsx

```diff
@@ -0,0 +1,59 @@
+import { useCallback } from 'react';
+import { useSelector } from 'src/hooks/suite';
+import {
+    CoinmarketPaymentMethodListProps,
+    CoinmarketPaymentMethodProps,
+    CoinmarketTradeBuySellType,
+    CoinmarketTradeDetailMapProps,
+} from 'src/types/coinmarket/coinmarket';
+import { CoinmarketPaymentMethodHookProps } from 'src/types/coinmarket/coinmarketForm';
+
+const useCoinmarketPaymentMethod = <
+    T extends CoinmarketTradeBuySellType,
+>(): CoinmarketPaymentMethodHookProps<T> => {
+    const paymentMethods = useSelector(state => state.wallet.coinmarket.info.paymentMethods);
+
+    const getPaymentMethods = (quotes: CoinmarketTradeDetailMapProps[T][]) => {
+        const newPaymentMethods: CoinmarketPaymentMethodListProps[] = [];
+
+        quotes.forEach(quote => {
+            const { paymentMethod } = quote;
+            const isNotInArray = !newPaymentMethods.some(item => item.value === paymentMethod);
+
+            if (typeof paymentMethod !== 'undefined' && isNotInArray) {
+                const label = quote.paymentMethodName ?? paymentMethod;
+
+                newPaymentMethods.push({ value: paymentMethod, label });
+            }
+        });
+
+        return newPaymentMethods;
+    };
+
+    const getQuotesByPaymentMethod = useCallback(
+        (
+            quotes: CoinmarketTradeDetailMapProps[T][] | undefined,
+            currentPaymentMethod: CoinmarketPaymentMethodProps,
+        ) => {
+            if (!quotes) return;
+
+            return quotes.filter(quote => {
+                if (currentPaymentMethod === '') return true; // all
```

Comment:
what about quote.error filtering in case of `all`?

Reply:
There will not be possibility to select all anymore and input quotes should be already filtered.

Reply:
So can we remove the line?